fs/littlefs/lfs_nuttx_defines.h

```diff
@@ -0,0 +1,43 @@
+/****************************************************************************
+ * fs/littlefs/lfs_nuttx_defines.h
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
```

Comment:
add process guard

Reply:
why? i don't expect this header to be included multiple times.

Reply:
do you mean this?
https://nuttx.apache.org/docs/latest/contributing/coding_style.html
```
Header File Idempotence. C header file must protect against multiple inclusion through the use of macros that “guard” against multiple definitions if the header file is included multiple times.
```

my interpretation is:
this header is not included multiple times. thus this rule doesn't apply.

Reply:
let's other people merge your change. I stop review this patch.